bmaptools/BmapHelpers.py

```diff
@@ -62,9 +62,10 @@ def get_block_size(file_obj):
 
     # Get the block size of the host file-system for the image file by calling
     # the FIGETBSZ ioctl (number 2).
-    binary_data = ioctl(file_obj, 2, struct.pack('I', 0))
-    bsize = struct.unpack('I', binary_data)[0]
-    if not bsize:
+    try:
+        binary_data = ioctl(file_obj, 2, struct.pack('I', 0))
+        bsize = struct.unpack('I', binary_data)[0]
+    except IOError as err:
```

Comment:
This would not work correctly in case if ioctl() returns successfully, but with bsize=0.

Reply:
good point.
is it possible bsize == 0 if ioctl returns successfully ? if yes we need another patch

Reply:
Yes, it is possible in some specific cases. e.g. AUFS in some docker setups.